src/WebJobs.Script.WebHost/Management/LinuxSpecialization/ManagedIdentityTokenProvider.cs

```diff
@@ -20,11 +20,11 @@ public class ManagedIdentityTokenProvider : IManagedIdentityTokenProvider
         private readonly IMetricsLogger _metricsLogger;
         private readonly ILogger<ManagedIdentityTokenProvider> _logger;
 
-        public ManagedIdentityTokenProvider(IEnvironment environment, HttpClient httpClient,
+        public ManagedIdentityTokenProvider(IEnvironment environment, IHttpClientFactory httpClientFactory,
             IMetricsLogger metricsLogger, ILogger<ManagedIdentityTokenProvider> logger)
         {
             _environment = environment;
-            _httpClient = httpClient;
+            _httpClient = httpClientFactory.CreateClient();
```

Comment:
Do we need to create the client this early in all cases? Could the client creation be deferred here?